scripts/babel-6/rewrite-modules.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+/**
+ * Rewrites module string literals according to the `_moduleMap` babel option.
+ * This allows other npm packages to be published and used directly without
+ * being a part of the same build.
+ */
+function mapModule(context, module) {
+  var moduleMap = context.state.opts._moduleMap || {};
+  if (moduleMap.hasOwnProperty(module)) {
+    return moduleMap[module];
+  }
+  // Jest understands the haste module system, so leave modules intact.
+  if (process.env.NODE_ENV !== 'test') {
+    var modulePrefix = context.state.opts._modulePrefix;
+    if (modulePrefix == null) {
+      modulePrefix = './';
+    }
+    return modulePrefix + module;
+  }
+}
+
+module.exports = function(babel) {
+
+  var t = babel.types;
+
+  /**
+   * Transforms `require('Foo')` and `require.requireActual('Foo')`.
+   */
+  function transformRequireCall(context, call) {
+    if (
+      !t.isIdentifier(call.callee, {name: 'require'}) &&
+      !(
+        t.isMemberExpression(call.callee) &&
+        t.isIdentifier(call.callee.object, {name: 'require'}) &&
+        t.isIdentifier(call.callee.property, {name: 'requireActual'})
+      )
+    ) {
+      return undefined;
+    }
+    var moduleArg = call.arguments[0];
+    if (moduleArg && moduleArg.type === 'Literal') {
+      var module = mapModule(context, moduleArg.value);
+      if (module) {
+        return t.callExpression(
+          call.callee,
+          [t.literal(module)]
+        );
+      }
+    }
+  }
+
+  /**
+   * Transforms either individual or chained calls to `jest.dontMock('Foo')`,
+   * `jest.mock('Foo')`, and `jest.genMockFromModule('Foo')`.
+   */
+  function transformJestCall(context, call) {
```

Comment:
Can you test this with the following code, and see what it produces?

```
jest.mock('foo').dontMock('foo').genMockFromModule('foo');
```

I get:

```
jest.mock('./././foo').dontMock('././foo').genMockFromModule('./foo');
```

Not that you broke it, but I'm pretty sure this never worked.